spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/EurekaClientConfigBeanTests.java

```diff
@@ -74,14 +74,14 @@ public void serviceUrlWithCompositePropertySource() {
 		this.context.getEnvironment().getPropertySources().addFirst(source);
 		source.addPropertySource(new MapPropertySource("config", Collections
 				.<String, Object> singletonMap("eureka.client.serviceUrl.defaultZone",
-						"http://example.com,http://example2.com")));
+						"http://example.com,http://example2.com, http://example3.com")));
 		this.context.register(PropertyPlaceholderAutoConfiguration.class,
 				TestConfiguration.class);
 		this.context.refresh();
-		assertEquals("{defaultZone=http://example.com,http://example2.com}",
+		assertEquals("{defaultZone=http://example.com,http://example2.com, http://example3.com}",
```

Comment:
I think I understand now, this test is really testing the `toString` result.  Where was the exception being thrown from?

Reply:
Without this fix, eureka server will throw the following exception when starting :
java.net.URISyntaxException: Illegal character in scheme name at index 0: http://eureka2:8762/eureka/
at java.net.URI$Parser.fail(Unknown Source) ~[na:1.8.0_121]
at java.net.URI$Parser.checkChars(Unknown Source) ~[na:1.8.0_121]
at java.net.URI$Parser.checkChar(Unknown Source) ~[na:1.8.0_121]
at java.net.URI$Parser.parse(Unknown Source) ~[na:1.8.0_121]
at java.net.URI.(Unknown Source) ~[na:1.8.0_121]
at com.netflix.eureka.cluster.PeerEurekaNodes.hostFromUrl(PeerEurekaNodes.java:258) [eureka-core-1.9.2.jar:1.9.2]
at com.netflix.eureka.cluster.PeerEurekaNodes.createPeerEurekaNode(PeerEurekaNodes.java:200) [eureka-core-1.9.2.jar:1.9.2]
...

And, the eureka client will throw the following exception when starting:
java.lang.IllegalArgumentException: Illegal character in scheme name at index 0:  http://bravo-eureka:8762/eureka/
	at java.net.URI.create(Unknown Source) ~[na:1.8.0_121]
	at com.sun.jersey.api.client.Client.resource(Client.java:434) ~[jersey-client-1.19.1.jar:1.19.1]
	at com.netflix.discovery.shared.transport.jersey.AbstractJerseyEurekaHttpClient.register(AbstractJerseyEurekaHttpClient.java:50) ~[eureka-client-1.9.2.jar:1.9.2]
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator$1.execute(EurekaHttpClientDecorator.java:59) [eureka-client-1.9.2.jar:1.9.2]
...

Reply:
The other option would be to do the `trim` at the point at which the exception is thrown.  You could make the argument that it might be better to avoid other bad manipulations of the url along the way.  I think this is fine though.  thanks!